File: my-solutions/0020-valid-parentheses/solution.py
class Solution:
    def isValid(self, s: str) -> bool:

        bracketStack = deque()
        for c in s:
            if len(bracketStack) == 0 and c in [')', '}', ']']:
                return False
            if len(bracketStack) > 0 and c in [')', '}', ']']:
                top = bracketStack[-1]
                if top == '(' and c != ')' or top == '{' and c != '}' or top == '[' and c != ']':
                    return False
                else:
                    bracketStack.pop()
            else:
                bracketStack.append(c)
        
        return len(bracketStack) == 0

File: my-solutions/0020-valid-parentheses/solution.cpp
class Solution {
public:
    bool isValid(string s) {
        stack<int> store;
        for(auto i = s.begin(); i != s.end(); i++) {
            if(*i == '(' || *i == '{' || *i == '[') {
                store.push(*i);
            } else if (*i == ')' || *i == '}' || *i == ']') {
                if(store.empty())
                    return false;
                
                char top = store.top();
                store.pop();
                if(top == '(' && *i != ')')
                    return false;
                else if(top == '{' && *i != '}')
                    return false;
                else if(top == '[' && *i != ']')
                    return false;
            } else {
                return false;
            }
        }
        if(!store.empty())
            return false;
        return true;
    }
};
